addons/account/models/account_move_line.py

```diff
@@ -1185,6 +1185,12 @@ def action_register_payment(self):
     # INVERSE METHODS
     # -------------------------------------------------------------------------
 
+    @api.onchange('tax_ids')
```

Comment:
This only works in the UI but won't work if the tax is removed programmatically

Reply:
and what is a better solution in that case?

Reply:
should I override the write method?

Reply:
The issue is probably something in `_sync_dynamic_lines` that doesn't work

Reply:
To get started, let's start by writing a test :wink: (not using the `Form` emulator)

Reply:
@william-andre I have wrote the unit test, plz have a look :)